src/components/Tags.tsx

```diff
@@ -0,0 +1,140 @@
+import * as React from 'react';
+import { MBTTAGS } from './TagSuggestions';
+
+//Dependencies for drag n drop tags with suggestions
+//npm install --save react-tag-input
+//npm install --save react-dnd@5.0.0
+//npm install --save react-dnd-html5-backend@3.0.2
+import { WithContext as ReactTags } from 'react-tag-input';
+import { stringify, v4 as uuid } from 'uuid';
+
+const suggestions = MBTTAGS.map(mbttags => {
+  return {
+    id: uuid(),
+    text: mbttags,
+  };
+});
+
+const KeyCodes = {
+  comma: 188,
+  enter: 13,
+};
+
+type tagstruct = {
+  rank: number;
+  segment: string;
+  tag: { createdAT: string; isDeleted: boolean; name: string; _id: string };
+  _id: string;
+};
+
+interface MyProps {
+  tags: { id: string; text: string }[];
+  seg: any;
+  rank: number;
+}
+
+interface MyState {
+  tags: any;
+  suggestions: any;
+  k: number;
+}
+
+const delimiters = [KeyCodes.comma, KeyCodes.enter];
+
+const capWords = (str: string) => {
+  if (!str) return 'Error';
+  let pieces = str.split(' ');
+  pieces.forEach((p, i) => (pieces[i] = p.charAt(0).toUpperCase() + p.substr(1)));
+  return pieces.join(' ');
+};
+
+class Tags extends React.Component<MyProps, MyState> {
+  constructor(props: MyProps) {
+    super(props);
+    this.state = {
+      tags: props.tags,
+      suggestions: suggestions,
+      k: 1,
+    };
+
+    this.handleDelete = this.handleDelete.bind(this);
+    this.handleAddition = this.handleAddition.bind(this);
+    this.handleDrag = this.handleDrag.bind(this);
+    this.handleTagClick = this.handleTagClick.bind(this);
+  }
+
+  handleDelete(i: any) {
+    const { tags } = this.state;
+    const delIndex = tags[i].id;
+
+    this.setState({
+      tags: tags.filter((tag: any, index: any) => index !== i),
+    });
+
+    //del from segment
+    for (let j = 0; j < this.props.seg.tags.length; j++) {
+      if (this.props.seg.tags[j].tag._id == delIndex) {
+        //console.log("Match ", i, delIndex, this.props.seg.tags);
+        this.props.seg.tags.splice(j, 1);
+      }
+      //console.log("Result ",i, this.props.seg.tags);
+    }
+  }
+
+  handleAddition(tag: any) {
+    tag.id = uuid();
+    tag.text = capWords(tag.text);
```

Comment:
I apologize if I have not mentioned this earlier, but we will soon be enforcing that all tags are all lower case.  This is part of an effort to normalize the tags and reduce/combine duplicate tags. I'm sure you have already noticed that both `Fear` and `fear` are individually in the top 10 most used tags. These need to reference the same tag, so we have to create a normalization scheme, and lowercasing all tags is the simplest/most bug free approach.

Frank K. is currently working on scripts to enforce this and to also retroactively enforce this rule on all existing tags.

Reply:
It is done. We force lowercase. 
I don't know what happens to tags like LCS or AUM ;-)